src/Core/Content/Product/Cms/ProductSliderCmsElementResolver.php

```diff
@@ -178,29 +172,24 @@ private function collectByProductStream(ResolverContext $resolverContext, FieldC
             $resolverContext->getSalesChannelContext()->getContext()
         );
 
-        $sorting = 'name:' . FieldSorting::ASCENDING;
-        $productStreamSorting = $elementConfig->get('productStreamSorting');
-        if ($productStreamSorting !== null) {
-            $sorting = $productStreamSorting->getStringValue();
-        }
-        $limit = self::FALLBACK_LIMIT;
-        $productStreamLimit = $elementConfig->get('productStreamLimit');
-        if ($productStreamLimit !== null) {
-            $limit = $productStreamLimit->getIntValue();
-        }
-
         $criteria = new Criteria();
         $criteria->addFilter(...$filters);
-        $criteria->setLimit($limit);
+        $criteria->setLimit($elementConfig->get('productStreamLimit')?->getIntValue() ?? self::FALLBACK_LIMIT);
         $criteria->addAssociation('options.group');
```

Comment:
$criteria->addAssociations(self::PRODUCT_ASSOCIATIONS);

Reply:
Thank you :-) I missed that place.